test/hotspot/jtreg/compiler/lib/ir_framework/test/IREncodingPrinter.java

```diff
@@ -176,6 +203,62 @@ private boolean hasAllRequiredFlags(String[] andRules, String ruleType) {
         return returnValue;
     }
 
+    private boolean hasRequiredFeaturesAnd(String[] andRules, String ruleType) {
+        for (int i = 0; i < andRules.length; i++) {
+            String feature = andRules[i];
+            i++;
+            String value = andRules[i];
+            TestFormat.check((value.contains("true") || value.contains("false")), "Incorrect value in " + ruleType + failAt());
+            if (!checkTargetFeature(feature, value)) {
+                // Rule will not be applied but keep processing the other flags to verify that they are same.
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean hasRequiredFeaturesOr(String[] orRules, String ruleType) {
```

Comment:
`ruleType` is always `applyIfTargetFeatureOr` and can be replaced as such.

Suggestion for method name: `hasAnyRequiredTargetFeature()` (I think the related existing `hasNoRequiredFlags()` method should be renamed/refactored to follow that convention as well but that's also for another day).